src/hooks/useSelectedRevisions.ts

```diff
@@ -23,22 +25,67 @@ const useSelectRevision = () => {
   const addSelectedRevisions = () => {
     const newSelected = [...selectedRevisions];
     newSelected.push(...baseCheckedRevisions, ...newCheckedRevisions);
-    //remove duplicates of the same revision
-    const filteredSelected = newSelected.filter(
-      (revision, index, self) =>
-        self.findIndex((r) => r.id === revision.id) === index,
-    );
+    //create a new array with unique values
+    const filteredSelected = [...new Set(newSelected)];
 
     dispatch(setSelectedRevisions({ selectedRevisions: filteredSelected }));
   };
 
+  const editSelectedRevisions = (searchType: InputType) => {
+    let revisionsForEdit = [...selectedRevisions];
+    switch (searchType) {
+      case 'base':
+        revisionsForEdit = [revisionsForEdit[0]];
+        break;
+      case 'new':
+        revisionsForEdit = revisionsForEdit.slice(1);
+        break;
+      default:
```

Comment:
if the default isn't possible, it would be good to throw an error so that we find programming errors easily.
(same below)

Reply:
Good idea